src/BotGroups.pas

```diff
@@ -0,0 +1,967 @@
+
+  setlength(multip2p.lovesek, length(multip2p.lovesek) + 1);
+  multip2p.lovesek[high(multip2p.lovesek)].fegyv := ownProps.fegyv;
+  multip2p.lovesek[high(multip2p.lovesek)].pos := state.pos;
+  multip2p.lovesek[high(multip2p.lovesek)].pos.y := state.pos.y + 1.5;
+  multip2p.lovesek[high(multip2p.lovesek)].v2 := tmpVec;
+  multip2p.lovesek[high(multip2p.lovesek)].kilotte:= -2;
+  state.canShoot := ownProps.baseLovesCooldown;
+end;
+
+procedure TBot.collideProjectile(loves: TLoves);
+var
+  lovesIndex: Integer;
+  isGun, amGun, isHit: boolean;
+  matWorld, matWorld2, matb:TD3DMatrix;
+  muks: TMuksoka;
+  tmp: TD3DXVector3;
+begin
+  if state.isDead > 0 then exit;
+
+  amGun := ownProps.fegyv < 128;
+  isGun := loves.fegyv < 128;
+  if isGun = amGun then exit;
+
+  muks := state.lastMuks;
+
+  isHit := FALSE;
+
+  D3DXMatrixRotationY(matWorld2, state.rotateY + D3DX_PI);
+  D3DXMatrixTranslation(matWorld, state.pos.x, state.pos.y, state.pos.z);
+  D3DXMatrixMultiply(matWorld2, matWorld2, matWorld);
+  case loves.fegyv of
+    FEGYV_M4A1:isHit:=-1 < _meglove(muks.gmbk, muks.kapcsk, loves.pos, loves.v2, 0.033, matWorld2);
+    FEGYV_M82A1:isHit:=-1 < _meglove(muks.gmbk, muks.kapcsk, loves.pos, loves.v2, 0.065, matWorld2);
+    FEGYV_MPG:isHit:=-1 < _meglove(muks.gmbk, muks.kapcsk, loves.pos, loves.v2, 0.15, matWorld2);
+    FEGYV_QUAD:isHit:=-1 < _meglove(muks.gmbk, muks.kapcsk, loves.pos, loves.v2, 0.2, matWorld2);
+    FEGYV_MP5A3:isHit:=-1 < _meglove(muks.gmbk, muks.kapcsk, loves.pos, loves.v2, 0.02, matWorld2);
+    FEGYV_BM3:isHit:=-1 < _meglove(muks.gmbk, muks.kapcsk, loves.pos, loves.v2, 0.085, matWorld2);
+    FEGYV_BM3_2:isHit:=-1 < _meglove(muks.gmbk, muks.kapcsk, loves.pos, loves.v2, 0.10, matWorld2);
+    FEGYV_BM3_3:isHit:=-1 < _meglove(muks.gmbk, muks.kapcsk, loves.pos, loves.v2, 0.13, matWorld2);
+    FEGYV_HPL:isHit:=-1 < _meglove(muks.gmbk, muks.kapcsk, loves.pos, loves.v2, 0.065, matWorld2);
+    FEGYV_GUNSUPP:isHit:=-1 < _meglove(muks.gmbk, muks.kapcsk, loves.pos, loves.v2, 0.033, matWorld2);
+    FEGYV_TECHSUPP:isHit:=-1 < _meglove(muks.gmbk, muks.kapcsk, loves.pos, loves.v2, 0.15, matWorld2);
+  end;
+
+  if isHit then
+  begin
+    state.isDead := ownProps.baseRespawnTime;
+    putRagdoll(state.pos, loves.v2);
+    exit;
+  end;
+end;
+
+procedure TBot.doLogic;
+const
+  gravity = 0.3;
+var
+  mapHeight: Single;
+  ojjektumCollideVec, directionVec, tmpVec: TD3DXVector3;
+  isOnSurface, isColliding, isBusy, hasTarget: boolean;
+  moveOnZ, moveOnX: boolean;
+  xSpeed, zSpeed: Single;
+  rotateYChange: Single;
+  tickNow: Cardinal;
+begin
+  tickNow := getTickCount;
+  if state.checkTargets > 0 then
+    state.checkTargets := state.checkTargets - 1;
+  if state.canShoot > 0 then
+    state.canShoot := state.canShoot - 0.01;
+  if state.isDead > 0 then
+  begin
+    state.isDead := state.isDead - 1;
+    exit;
+  end
+  else if state.isDead = 0 then
+  begin
+    state.isDead := -1;
+    //state.pos := ownProps.respawnPos;
```

Comment:
Ez nem vagyok benne biztos h miert lett kiveve, de amikor legutobb neztem mukodott minden. Akarhogy is, ennek elvileg nem kene kikommentelve lennie.